MiniZincIDE/highlighter.h

```diff
@@ -22,6 +22,123 @@ struct Bracket {
     int pos;
 };
 
+class ThemeColor{
+public:
+    QColor light;
+    QColor dark;
+    ThemeColor(){
+        light = Qt::red;
+        dark = Qt::red;
+    }
+
+    ThemeColor(QColor lightAndDark){
+        light = lightAndDark;
+        dark = lightAndDark;
+    }
+
+    ThemeColor(QColor l, QColor d){
+        light = l;
+        dark = d;
+    }
+    QColor get(bool darkMode){
+        return darkMode?dark:light;
+    }
+};
+
+struct Theme
+{
+        ThemeColor textColor, // Color of normal text
+        keywordColor, // Color of MiniZinc keywords such as "in" "where" "constraints"
+        functionColor, // Color of words that are followed by a function call, such as "forall()"
+        stringColor, // Color of strings
+        commentColor, // Color of comments %
+        backgroundColor, // Color of background
+        errorColor, //Color of error messages
+        lineHighlightColor,  // Color of highlighted line
+        foregroundActiveColor, // color of active line number
+        foregroundInactiveColor, // color of inactive line number
+        textHighlightColor,      // color of highlighted text
+        bracketsMatchColor, // Color of matching brackets
+        bracketsNoMatchColor //Colors of dangling brackets
+        ;
+        Theme(): Theme(Qt::white,Qt::darkGreen, Qt::blue, Qt::darkRed, Qt::red, Qt::white, Qt::lightGray) {}
+
+        Theme (QColor _textColor,
+               QColor _keywordColor,
+               QColor _functionColor,
+               QColor _stringColor,
+               QColor _commentColor,
+               QColor _backgroundColor,
+               QColor _lineHighlightColor):
+            Theme(ThemeColor(_textColor,_textColor),
+            ThemeColor(_keywordColor,_keywordColor),
+            ThemeColor(_functionColor,_functionColor),
+            ThemeColor( _stringColor,_stringColor),
+            ThemeColor(_commentColor,_commentColor),
+            ThemeColor(_backgroundColor,_backgroundColor),
+            ThemeColor(_lineHighlightColor,_lineHighlightColor),
+            ThemeColor(Qt::red,Qt::red))
+            {}
+
+        Theme(ThemeColor _textColor,
```

Comment:
While the calls constructing a Theme look ok in an IDE such as CLion that can annotate each argument with the parameter name, it is less clear in less capable IDEs or editors.

Unfortunately, C++ doesn't have keyword arguments, and none of the alternatives are really that great. So unclear to me if there are any really better ways to handle this.

Reply:
I agree on both points, but I cannot see a cleaner way of doing this.

The good news is that themes are rarely defined in the code, so not knowing which argument in the constructor is which is a rarely an issue. Also, C++20 does support constructors with named parameters, so this can easily be fixed in the future if the source migrates to C++20.

Reply:
Yes, that will be a nice addition to have. And the calls in the code are well-documented, so not a large problem at all!